src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/shim.cxx

```diff
@@ -1036,10 +1052,17 @@ namespace xclhwemhal2 {
         //if (!xclemulation::file_exists(sim_file))
         if (!boost::filesystem::exists(sim_file))
           sim_file = "simulate.sh";
+          
+        if (mLogStream.is_open() )
+            mLogStream << __TIME__ <<"\t"<< __func__ << " The simulate script is "  <<sim_file  << std::endl;
 
         int r = execl(sim_file.c_str(), sim_file.c_str(), simMode, NULL);
-        fclose(stdout);
         if (r == -1){ std::cerr << "FATAL ERROR : Simulation process did not launch" << std::endl; exit(1); }
+
+        if (mLogStream.is_open() )
+            mLogStream << __TIME__ <<"\t"<< __func__ << " After launching xsim process"  << std::endl;
+
+        fclose(stdout);
         exit(0);
```

Comment:
Really?

Reply:
I am not necessarily following the flow, but exit() just doesn't seem right.  What would happen if you just throw an exception, won't the application get a chance to catch and exit gracefully?    What if host application is don't more than just emulation, do you think Simulation not started should bring down the whole process?